pages/blog/unleash-the-power-of-fragments-with-graphql-codegen.mdx

```diff
@@ -0,0 +1,314 @@
+import withArticle from '../../ui/blog/article'
+
+export const meta = {
+  title: 'Unleash the power of Fragments with GraphQL Codegen',
+  author: 'laurin',
+  tags: [
+    'graphql',
+    'codegen',
+    'react',
+    'typescript',
+    'relay',
+    'apollo',
+    'urql'
+  ],
+  date: '2022-08-03',
+  description:
+    "The most important parts of Relay are the concepts of building and scaling applications, let's show how you can use these patterns in your existing project, no matter what client library you are currently using.",
+  image:
+    '/blog-assets/unleash-the-power-of-fragments-with-graphql-codegen/cover.png',
+  thumbnail:
+    '/blog-assets/unleash-the-power-of-fragments-with-graphql-codegen/thumbnail.png'
+}
+
+export default withArticle(meta)
+
+There is no doubt that Relay is the most advanced JavaScript GraphQL Client that is out there. However, at the same time, the learning curve and adoption is yet very sparse compared to other popular alternatives such as Apollo Client, Urql, or GraphQL Request.
+
+As a consequence, many people don't even know about all the benefits and patterns used within (and so far being exclusive to) Relay.
+The Guild gathered and worked on projects whose GraphQL client usage spans across all the available solutions today.
+In our opinion, the most important parts of Relay are the concepts of building and scaling applications.
+These patterns are actually applicable to any GraphQL Client, you just need the right code generation tool for the job.
+
+On a high level these benefits are the following:
+
+**A fragmentized component tree.** Colocating the GraphQL operation and fragment definitions with the components and composing them up for only sending a single request to our server.
```

Comment:
here you say what they concepts are, you don't mention their benefits and why they are beneficial

Reply:
done